### ISSUE TITLE
overfitting when crop data

### ISSUE THREAD

**Issue by butterflyro** [MIC-DKFZ/nnUNet#782]
Hi FabianIsensee, Thank you so much for what you do for everyone.

I'm having a problem with a training model.

I have 113 cases of brain aneurysm segmentation data. The dataset has 3 classes: 0: background, 1: untreated unruptured aneurysm, 2: treated aneurysm, or morphology from treated aneurysm. And need to recognize label 1. The data set has 4 cases of width and height less than 400, 101 cases greater than 400 less than 800, 8 cases greater than 800. When I train this data without changing change anything. Well, the training is pretty good as shown in Figure 1. Due to data imbalance, Foreground vs background: 8.3x10^-6 (Figure 2). I try to crop the Circle of Willis region by width and height, the depth stays the same, but the result is looks worse (Figure 3).

![image](https://user-images.githubusercontent.com/64973267/134118774-9f627cad-7611-4d11-a529-181c373bba7f.png)
Figure 1

![image](https://user-images.githubusercontent.com/64973267/134118821-31992e96-ed75-4990-ac3d-217965a22d13.png)
Figure 2

![image](https://user-images.githubusercontent.com/64973267/134118901-d7ccb5a4-0c38-4a7f-9f67-c89e7dddaac2.png)
Figure 3

Can you explain to me why it's worse when i remove the unnecessary areas?
Can you give me some advice when encountering such imbalanced data?

**Comment by FabianIsensee** [comment]
Hi, thanks for sharing. nnU-Net deals with imbalanced data quite nicely, there is nothing you need to do. Still, its performance should not break down that much when cropping to a smaller region. Have you checked where most of the errors occur? Maybe it has problems distinguishing label 1 and 2 in these scenarios because it does not see enough contextual information
Also try the `nnUNetTrainerV2_Loss_DiceCE_noSmooth` trainer form the current master and let me know how it goes :-)

**Comment by FabianIsensee** [comment]
> The second thing i can ask is: When data augmentation for the train set, my val doesn't have the same variations as when augumentation the train set, i think it will give a bad result, right.

No this is the correct way of doing data augmentation. The validation set needs to be representative of the test cases so it cannot be augmented (we could accidentally make it very different and thus no longer representative). What we do with the training dataset is up to us and we can modify it as mcuh as we want

> Can you explain me a little bit how nnUNet handles imbalance, I read the paper and it says using oversampling technique, i don't think with only one technique it will handle good.

Disagree. nnU-net's oversampling strategy is very robust and there is no evidence that it isn't enough. If you have such evidence then I am of course happy to take a look :-)
The way nnU-Net does oversampling is by sampling 66% of crops randomly and the remaining 33% by first picking a foreground class and then picking a location where this class exists.

My advice was not to add smooth to the Dice loss but rather to remove it. Your progress plot looks way better than the one you initially posted where the green line was dropping hard - how is the new model worse than the old one?
